tests/functional/WSP/test_wsp.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015, 2016, 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Functional tests for WSP spider"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import pytest
+import json
+import os
+
+from itertools import islice
+from scrapyd_api import ScrapydAPI
+from time import sleep
+
+from tests.functional.tasks import app
+
+
+class CeleryMonitor(object):
+    def __init__(self, app, monitor_timeout=3, monitor_iter_limit=100):
+        self.results = []
+        self.recv = None
+        self.app = app
+        self.connection = None
+        self.monitor_timeout = monitor_timeout
+        self.monitor_iter_limit = monitor_iter_limit
+
+    def __enter__(self):
+        state = self.app.events.State()
+
+        def announce_succeeded_tasks(event):
+            state.event(event)
+            task = state.tasks.get(event['uuid'])
+            print('TASK SUCCEEDED: %s[%s] %s' % (task.name, task.uuid, task.info(),))
+            tasks = app.AsyncResult(task.id)
+            for task in tasks.result:
+                self.results.append(task)
+            self.recv.should_stop = True
+
+        def announce_failed_tasks(event):
+            state.event(event)
+            task = state.tasks.get(event['uuid'])
+            print('TASK FAILED: %s[%s] %s' % (task.name, task.uuid, task.info(),))
+            self.results.append(task.info())
+            self.recv.should_stop = True
+
+        self.app.control.enable_events()
+        self.connection = self.app.connection()
+        self.recv = self.app.events.Receiver(self.connection, handlers={
+            'task-succeeded': announce_succeeded_tasks,
+            'task-failed': announce_failed_tasks,
+        })
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        events_iter = self.recv.itercapture(limit=None, timeout=self.monitor_timeout, wakeup=True)
+        self._wait_for_results(events_iter)
+        self.connection.__exit__()
+
+    def _wait_for_results(self, events_iter):
+        any(islice(events_iter, self.monitor_iter_limit))
+
+    @classmethod
+    def do_crawl(cls,
```

Comment:
Here try to stick to the style you were using, newline right after the opening parenthesis, only one indentation level, then newline before the closing parenthesis, that it's dedented to match the opening parenthesis indentation:

```python
def bla(
    one,
    two,
)
```